### ISSUE TITLE
"Undefined index: choice_list" with Symfony 3.0

### ISSUE THREAD

**Issue by sbooker** [sonata-project/SonataAdminBundle#3529]
I use `sonata_type_model` for many to many relation.
```PHP
class UserAdmin extends Admin {

protected function configureFormFields(FormMapper $form) 
{
            $form
                ->add(
                    'users',
                    'sonata_type_model',
                    array(
                        'label' => 'Users',
                        'by_reference' => false,
                        'multiple' => true,
                        'expanded' => true,
                        'required' => false,
                        'query' => $this->getUsersQueryBuilder
                    )
                )
            ;
}
}
```
And have error "Undefined index: choice_list". Stack trace:
```PHP
Stack Trace
in vendor/sonata-project/admin-bundle/Form/Type/ModelType.php at line 43   -
    {
        if ($options['multiple']) {
            if (array_key_exists('choice_loader', $options) && $options['choice_loader'] !== null) { // SF2.7+
                $builder->addViewTransformer(new ModelsToArrayTransformer($options['choice_list'], $options['model_manager'], $options['class']), true);
            } else {
                $builder->addViewTransformer(new LegacyModelsToArrayTransformer($options['choice_list']), true);
            }
```
As I can see `$options['choice_list']` never not initialized for Symfony 2.7+.
I offer change line 43 in `sonata-project/admin-bundle/Form/Type/ModelType.php` to 
```PHP
                $builder->addViewTransformer(
                    new ModelsToArrayTransformer(
                        new LazyChoiceList($options['choice_loader']),
                        $options['model_manager'],
                        $options['class']),
                    true
                );
```

**Comment by ajgarlag** [comment]
@sbooker Your proposed patch works form me. Maybe you could submit a PR.

**Comment by sbooker** [comment]
@ajgarlag PR: https://github.com/sonata-project/SonataAdminBundle/pull/3546

**Comment by wslaghekke** [comment]
This fixes the problem for me,
Any clue when or if this will get merged?

**Comment by greg0ire** [comment]
Fixed by #3655
